presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -115,9 +117,21 @@ public HttpRequestSessionContext(HeaderSupport forwardedHeaderSupport, HttpServl
         path = trimEmptyToNull(servletRequest.getHeader(PRESTO_PATH));
         assertRequest((catalog != null) || (schema == null), "Schema is set but catalog is not");
 
+        String authenticatedUser = (String) servletRequest.getAttribute(PRESTO_USER);
+        if (authenticatedUser != null) {
```

Comment:
I don't like the fact, when there is no extracted user from principal we do not pass anything to access control. There might be access control that could verify if we can impersonate the user basing only on principal (like checkCanSetUser did).